src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -530,6 +523,24 @@ private void getCanvasResponse(HttpServletRequest request,
 		out.flush();
 	}
 
+	public String unmaskUrl(String url, TcSessionData tc,
+			String stringToReplaceUser, String stringToReplaceCourse) {
+		if (tc != null){
+			String uniqname = (String) tc.getCustomValuesMap().get(CUSTOM_CANVAS_USER_LOGIN_ID);
+			String courseId = (String) tc.getCustomValuesMap().get(CUSTOM_CANVAS_COURSE_ID);
+			M_log.debug("uniqname: " + uniqname);
+			String replaceValue = "as_user_id=sis_login_id:" + uniqname;
+			if(url.toLowerCase().contains(stringToReplaceUser.toLowerCase())){
+				url = url.replace(stringToReplaceUser.toLowerCase(), replaceValue);
+			}
+			if(url.toLowerCase().contains(stringToReplaceCourse.toLowerCase())){
+				url = url.replace(stringToReplaceCourse.toLowerCase(), courseId);
+			}
+			M_log.debug("New URL: " + url);			
```

Comment:
I would give a descriptive name to the debug messages, this would be helpful if debugging on an issues. I see a lot of  text saying URL around the code,  while debugging it would be helpful what this URL means in various lines in logs